html/includes/api_functions.inc.php

```diff
@@ -506,6 +506,31 @@ function list_bgp()
 }
 
 
+function get_bgp()
+{
+    check_is_read();
+
+    $app        = \Slim\Slim::getInstance();
+    $router = $app->router()->getCurrentRoute()->getParams();
+
+    $bgpPeerId        = $router['id'];
+    if (!is_numeric($bgpPeerId)) {
+        api_error(400, 'Invalid id has been provided');
+    }
+
+    $bgp_session       = dbFetchRows("SELECT `bgpPeers`.* FROM `bgpPeers` LEFT JOIN `devices` ON `bgpPeers`.`device_id` = `devices`.`device_id` WHERE `bgpPeerState` IS NOT NULL AND `bgpPeerState` != '' AND bgpPeer_id = ?", array($bgpPeerId));
```

Comment:
Why are you joining the devices table here if you only return data from the bgpPeers table?

Reply:
My bad, this was copied and adapted from the list_bgp() function. Will fix shortly.

Reply:
Fixed for you :)